jetty-server/src/main/java/org/eclipse/jetty/server/session/FileSessionDataStore.java

```diff
@@ -206,21 +208,24 @@ public void sweepDisk()
         long now = System.currentTimeMillis();
         if (LOG.isDebugEnabled())
             LOG.debug("Sweeping {} for old session files", _storeDir);
-        try
+        try (Stream<Path> stream = Files.walk(_storeDir.toPath(), 1, FileVisitOption.FOLLOW_LINKS))
         {
-            Files.walk(_storeDir.toPath(), 1, FileVisitOption.FOLLOW_LINKS)
+            stream
                 .filter(p -> !Files.isDirectory(p)).filter(p -> !isOurContextSessionFilename(p.getFileName().toString()))
                 .filter(p -> isSessionFilename(p.getFileName().toString()))
+                .collect(Collectors.toList()) // Don't delete whilst walking
                 .forEach(p ->
                 {
-
                     try
                     {
-                        sweepFile(now, p);
+                        if (!Files.deleteIfExists(p))
```

Comment:
Bzzzt. Do not pass go, do not collect $100. YOu can only delete it if it has expired!

Reply:
oooops fixed